stix2/properties.py

```diff
@@ -325,7 +330,7 @@ def clean(self, value):
         if self.type:
             if not value.startswith(self.type):
                 raise ValueError("must start with '{0}'.".format(self.type))
-        if not REF_REGEX.match(value):
+        if not ID_REGEX.match(value):
             raise ValueError("must match <object-type>--<guid>.")
```

Comment:
Now that we are forcing UUIDv4 we could change this message to: "must match <object-type>--<uuidv4>." since guid is kind of general...